appsensor-core/src/main/java/org/owasp/appsensor/core/configuration/server/ServerConfiguration.java

```diff
@@ -190,6 +205,47 @@ public ServerConfiguration setGeolocationDatabasePath(String geolocationDatabase
 		return matches;
 	}
 	
+	/**
+	 * Locate matching client name in custom configuration from server-side config file. 
+	 * 
+	 * @param clientApplicationName the client name for which the detection point exists
+	 */
+	public Boolean findCustomDectectionClientName(String clientName){
+	if(getCustomDetectionPoints().size() != 0){
+			return getCustomDetectionPoints().containsKey(clientName);
+		}
+		return false;
+	
+	}
+	
+	/**
+	 * Locate matching detection points configuration from server-side config file. 
+	 * 
+	 * @param search detection point that has been added to the system
+	 * @param clientApplicationName the client name for which the detection point exists
+	 * @return DetectionPoint populated with configuration information from server-side config
+	 */
+	
+	public Collection<DetectionPoint> findDetectionPoints(DetectionPoint search, String clientApplicationName) {
+			Collection<DetectionPoint> matches = new ArrayList<DetectionPoint>();
+			
+			for (Entry customDetectionPoint : getCustomDetectionPoints().entrySet()) {
+			String clientName = customDetectionPoint.getKey().toString();			
+			if(clientApplicationName.equals(clientName)){
+			   List<DetectionPoint> customList = (List<DetectionPoint>)customDetectionPoint.getValue();
+			   for(DetectionPoint customPoint: customList)
+			   {
+					if (customPoint.typeMatches(search)) {
+						matches.add(customPoint);
+					}
+				}
+			}
+			
+			
```

Comment:
@shreyasdn the way I would do this is keep 2 lists. The first list would have what you've already done in this method. The second list would come from calling the existing findDetectionPoints method. That would give you the "normal" detection points, and the "custom" detection points for this client in 2 different lists. I would then "merge" the 2 lists by adding all the "custom" detection points first. Then I would go through the "normal" detection points and add each one ONLY if it does not already exist. Let me know if this is unclear.